handlers/common.go

```diff
@@ -37,6 +40,57 @@ func getOptionsAndRedirect(form url.Values, redirectURI *string) (options []stri
 	return options
 }
 
+// getIDNameLookupFromDatasetAPI creates a map of option keys and labels from the provided filter options,
+// getting the labels from DatasetAPI with paginated calls until all values have been found.
+// Note that this method may be expensive, if you can get the labels from some other available source, it would be preferred.
```

Comment:
Yes, that was an idea we also had (while working on the hierarchy), but Eleonor mentioned that it's not very 'restful'. Then we found that the hierarchy model contained labels, but we didn't have labels for all the items, so we ended up using this.
Maybe we can consider accepting as list of IDs in dataset API in the future if we see that the performance is not good enough.
Alternatively, if we don't want to pass a list, we could consider doing a binary search instead of sequential batches (it would be a big performance improvement if len(filterOptions) << len(datasetOptions), otherwise I think that wouldn't be a huge improvement)